apps/frontend/src/app/5_pages/MarketMakingPage/components/AdjustAndDepositModal/AdjustAndDepositModal.tsx

```diff
@@ -52,35 +62,120 @@ export const AdjustAndDepositModal: FC<AdjustAndDepositModalProps> = ({
 }) => {
   const [adjustType, setAdjustType] = useState(AdjustType.Deposit);
   const [value, setValue, amount] = useWeiAmountInput('');
+  const [tokenPoolBalance, setTokenPoolBalance] = useState('0');
+  const { account } = useAccount();
+  const { onDeposit, onWithdraw } = useHandleMarketMaking(onClose);
+  const { balanceA, loadingA } = useGetUserInfo(pool);
+
+  const poolWeiAmount = useMemo(
+    () =>
+      decimalic(amount.toString())
+        .div(decimalic(balanceA.toString()).div(tokenPoolBalance))
```

Comment:
```suggestion
        .div(balanceA).div(tokenPoolBalance))
```
This is an example of incorrect typing - `balanceA` is already `Decimal` but because the typings in `useGetUserInfo` are incorrect, `balanceA` type is shown as `any`. I am gonna leave a comment with instructions on how to fix it in `useGetUserInfo` but make sure that you go through every usage and fix it so we don't have unnecessary conversions.